### PR TITLE
CLOUDP-186825: AtlasProject controller to watch secrets for alert configurations

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

AtlasProject controller to watch secrets for alert configurations

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/controller/atlasproject/alert_configurations.go [hunk 1]
```diff
@@ -48,54 +50,69 @@ func (r *AtlasProjectReconciler) ensureAlertConfigurations(service *workflow.Con
...
 
 // This method reads secrets refs and fills the secret data for the related Notification
-func readAlertConfigurationsSecretsData(kubeClient client.Client, projectNs string, alertConfigs []mdbv1.AlertConfiguration) error {
+func (r *AtlasProjectReconciler) readAlertConfigurationsSecretsData(project *mdbv1.AtlasProject, alertConfigs []mdbv1.AlertConfiguration) error {
+	resourcesToWatch := make([]watch.WatchedObject, 0)
+	projectNs := project.Namespace
+	defer func() {
+		r.EnsureMultiplesResourcesAreWatched(types.NamespacedName{Namespace: project.Namespace, Name: project.Name}, r.Log, resourcesToWatch...)
+		r.Log.Debugf("watching alert configuration secrets: %v\r\n", r.WatchedResources)
+	}()
```
Comment (josvazg) [inline pkg/controller/atlasproject/alert_configurations.go:59]: Why is a defer function needed, rather than just putting this code after the for loop?

Comment (igor-karpukhin) [reply pkg/controller/atlasproject/alert_configurations.go:59]: Yes, the operator should watch for a specified secret even if there is no fields for a specific alert configuration. E.g: user configures one AtlasProject instance with an alert configuration using secret. The secret is missing the required data. If we won't add this secret to watcher, changes to this secret won't trigger reconciliation of a Project. I'll update the code to match that.

Comment (josvazg) [reply pkg/controller/atlasproject/alert_configurations.go:59]: ok

**Review by josvazg — APPROVED** [review]

**Review by helderjs — APPROVED** [review]

**Comment by josvazg** [comment]
LGTM

**Comment by igor-karpukhin** [comment]
@josvazg @helderjs @roothorp  please take a look when you have time. I have slightly modified the resource watcher

**Pull request merged**